Orange/canvas/application/settings.py

```diff
@@ -359,6 +360,24 @@ def __setupUi(self):
 
         tab.setLayout(form)
 
+        # Error Reporting Tab
+        tab = QWidget()
+        self.addTab(tab, self.tr("Error Reporting"),
+                    toolTip="Settings related to error reporting")
+
+        class MachineIDLineEdit(QLineEdit):
+            def __init__(self):
+                super().__init__()
+                self.setText(get_machine_id())
+
+            def focusOutEvent(self, QFocusEvent):
+                super().focusOutEvent(QFocusEvent)
+                set_machine_id(self.text())
+
+        form = QFormLayout()
+        form.addRow("Machine ID:", MachineIDLineEdit())
+        tab.setLayout(form)
```

Comment:
I was thinking about that but those settings are stored under the version directory and are hence reset with each new version.

If there are other things that should persist through versions we should probably develop something more general (like those settings above) that persists through updates.